core/src/main/scala/org/apache/spark/sql/execution/row/RowDeleteExec.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2017 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package org.apache.spark.sql.execution.row
+
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, ExprCode, ExpressionCanonicalizer}
+import org.apache.spark.sql.catalyst.expressions.{Attribute, BindReferences}
+import org.apache.spark.sql.execution.{SparkPlan, TableMutationExec}
+import org.apache.spark.sql.sources.ConnectionProperties
+import org.apache.spark.sql.store.CodeGeneration
+import org.apache.spark.sql.types.{StructField, StructType}
+
+/**
+ * Generated code plan for deletes in a row table.
+ */
+case class RowDeleteExec(child: SparkPlan, resolvedName: String,
+    tableSchema: StructType, keyColumns: Seq[Attribute],
+    connProps: ConnectionProperties, onExecutor: Boolean)
+    extends TableMutationExec(forUpdate = false) {
+
+  @transient private var result: String = _
+  @transient private var stmt: String = _
+  @transient private var batchCount: String = _
+
+  override protected def doProduce(ctx: CodegenContext): String = {
+    val (initCode, commitCode, endCode) = connectionCodes(ctx)
+    result = ctx.freshName("result")
+    ctx.addMutableState("long", result, s"$result = -1L;")
+    stmt = ctx.freshName("statement")
+    batchCount = ctx.freshName("batchCount")
+    val childProduce = doChildProduce(ctx)
+    s"""
+       |if ($result >= 0L) return;
+       |$initCode
+       |try {
+       |  final java.sql.PreparedStatement $stmt = $connTerm.prepareStatement(
+       |    "${s"DELETE FROM $resolvedName WHERE $whereClauseString"}");
```

Comment:
Are you saying something like df.write.update()? What does "updating" a DataFrame into a table mean? An insertInto or putInto of a DataFrame makes sense but not update. This makes sense as a Dataset API.

A deleteFrom like what has been added might make sense saying "that delete all rows of given Dataset from target table". However, as of now it is not of much use except for CDC kind of scenario since it requires all the columns of source to match target. This is an unrealistic expectation and needs to be fixed to allow for differing columns and a way to provide column mapping from source to target.